File: Array & String/Remove Duplicates from Sorted Array/RemoveDuplicates.go
func removeDuplicates(nums []int) int {
	uniqueMap := make(map[int]bool)
	var uniqueSlice []int

	for _, num := range nums {
		if !uniqueMap[num] {
			uniqueMap[num] = true
			uniqueSlice = append(uniqueSlice, num)
		}
	}

	copy(nums, uniqueSlice)
	return len(uniqueSlice)
}

File: Array & String/Remove Duplicates from Sorted Array II/RemoveDuplicates.rb
def remove_duplicates(nums)
  return nums.length if nums.length <= 2
  unique_count = 2
  (2...nums.length).each do |i|
    if nums[i] != nums[unique_count - 2]
      nums[unique_count] = nums[i]
      unique_count += 1
    end
  end
  unique_count
end
